timelock-impl/src/main/java/com/palantir/atlasdb/timelock/batch/MultiClientConjureTimelockResource.java

```diff
@@ -111,6 +117,44 @@ public ListenableFuture<Map<Namespace, GetCommitTimestampsResponse>> getCommitTi
                 MoreExecutors.directExecutor()));
     }
 
+    @Override
+    public ListenableFuture<Map<Namespace, ConjureUnlockResponseV2>> unlock(
+            AuthHeader authHeader, Map<Namespace, ConjureUnlockRequestV2> requests) {
+        return handleExceptions(() -> Futures.transform(
+                Futures.allAsList(Collections2.transform(
+                        requests.entrySet(), e -> unlockForSingleNamespace(e.getKey(), e.getValue()))),
+                ImmutableMap::copyOf,
+                MoreExecutors.directExecutor()));
+    }
+
+    private ListenableFuture<Entry<Namespace, ConjureUnlockResponseV2>> unlockForSingleNamespace(
```

Comment:
There is a bit of overhead here that could be removed, but since this is not a hot codepath and it makes the code nicer to read, it's fine